### DISCUSSION TITLE
Alternating 2nd frequency for modes

### DISCUSSION THREAD

**Discussion by whallmann** [mikaelnousiainen/RS41ng#57] [Ideas]
We do have here in europe some areas where the RX parked on different  frequencies e.g  PA, ON, DE  434.713, e.g poland and balkan countries on 437.6 MHz.  To get the most effort on flying here, it would be nice to transmitt on both . So i prepared some code to radio.c and config.h so each transmittion slot alternates between the two frequencies. Worked excellent on an 2 day flight from DL to Greece. I ask the crowd now if this is interessting to add to standard?
cu/73 Wolf DF7PN

**Comment by whallmann** [comment]
remark: done this only for 4fsk v2.

**Reply by mikaelnousiainen** [reply]
@whallmann Please file a pull request if you wish to have new features included.

However, in a way it could be better if this sort of (not so often needed) feature was included in the documentation -> maybe write a short guide on how to add extra TX frequencies/modes on radio.c / config.h?

After all, the radio.c list of transmit entries can be used to create arbitrarily complex transmission patterns with any set of frequencies, modes and schedules.

**Comment by dalewegner** [comment]
Trying to do the same thing for a flight from USA to Europe and would like the broadest coverage possible because crossing into European area could be in areas that use two different frequencies.  Could you attached the config.h and radio.c files with code modifications to this thread now and follow-up with documentation later? 73 KB3ZOX

**Reply by mikaelnousiainen** [reply]
In `radio.c` file, find the following place: (assuming here you're transmitting Horus 4FSK v2)

```
#if RADIO_SI4032_TX_HORUS_V2
        {
                .enabled = RADIO_SI4032_TX_HORUS_V2,
                .radio_type = RADIO_TYPE_SI4032,
                .data_mode = RADIO_DATA_MODE_HORUS_V2,
                .transmit_count = RADIO_SI4032_TX_HORUS_V2_COUNT,
                .time_sync_seconds = HORUS_V2_TIME_SYNC_SECONDS,
                .time_sync_seconds_offset = HORUS_V2_TIME_SYNC_OFFSET_SECONDS,
                .frequency = RADIO_SI4032_TX_FREQUENCY_HORUS_V2,
                .tx_power = RADIO_SI4032_TX_POWER,
                .symbol_rate = HORUS_V2_BAUD_RATE_SI4032,
                .payload_encoder = &radio_horus_v2_payload_encoder,
                .fsk_encoder_api = &mfsk_fsk_encoder_api,
        },
#endif
```

You can add a second entry there with a different frequency:

```
#if RADIO_SI4032_TX_HORUS_V2
// Define another frequency here (or move it into config.h, doesn't matter)
#define RADIO_SI4032_TX_FREQUENCY_2_HORUS_V2 432500000
        {
                .enabled = RADIO_SI4032_TX_HORUS_V2,
                .radio_type = RADIO_TYPE_SI4032,
                .data_mode = RADIO_DATA_MODE_HORUS_V2,
                .transmit_count = RADIO_SI4032_TX_HORUS_V2_COUNT,
                .time_sync_seconds = HORUS_V2_TIME_SYNC_SECONDS,
                .time_sync_seconds_offset = HORUS_V2_TIME_SYNC_OFFSET_SECONDS,
                .frequency = RADIO_SI4032_TX_FREQUENCY_HORUS_V2, // This is the first frequency
                .tx_power = RADIO_SI4032_TX_POWER,
                .symbol_rate = HORUS_V2_BAUD_RATE_SI4032,
                .payload_encoder = &radio_horus_v2_payload_encoder,
                .fsk_encoder_api = &mfsk_fsk_encoder_api,
        },
        {
                .enabled = RADIO_SI4032_TX_HORUS_V2,
                .radio_type = RADIO_TYPE_SI4032,
                .data_mode = RADIO_DATA_MODE_HORUS_V2,
                .transmit_count = RADIO_SI4032_TX_HORUS_V2_COUNT,
                .time_sync_seconds = HORUS_V2_TIME_SYNC_SECONDS,
                .time_sync_seconds_offset = HORUS_V2_TIME_SYNC_OFFSET_SECONDS,
                .frequency = RADIO_SI4032_TX_FREQUENCY_2_HORUS_V2, // This is the second frequency
                .tx_power = RADIO_SI4032_TX_POWER,
                .symbol_rate = HORUS_V2_BAUD_RATE_SI4032,
                .payload_encoder = &radio_horus_v2_payload_encoder,
                .fsk_encoder_api = &mfsk_fsk_encoder_api,
        },
#endif
```

**Reply by mikaelnousiainen** [reply]
@dalewegner Out of curiosity, how are you planning to fly a floater (or a long flight in general) with RS41 hardware? What do you use to power the sonde?

**Reply by whallmann** [reply]
Yes i first thought about to add another block in the queue with a 2nd frequency. I decided aginst this solution because the user have to handle within the code of radio.c and should get it easy enabled/disabled - i used the config.h.   So i realized my method you should see. 
I think i have to fork it first, will try this way.

your remarks how to power a rs41 floater. Read [discussion](https://github.com/mikaelnousiainen/RS41ng/discussions/63)  please

**Comment by dalewegner** [comment]
Thanks.  Horus is what we wanted to use since that is the only existing
network in Europe on 70 CM.  Since the RS41 radio is not able to access
APRS on 2 M that is the only option.  Satellite coverage on 70 CM does not
seem to exist.  And WSPR on 70 CM does not have very long range.  Goal is
not to add boards to RS41, but use what we have.

I am sure it will take a few attempts.  The winter winds some days are very
strong.  Previous attempts have been within 300 miles of Europe.  A low
accent/descent latex, multiple latex at one time bouncing off ocean as each
bursts to stay lower in atmosphere, and third with homemade mylar balloon.
The crossing should be able to be done on battery with good power
management settings or extra batteries.  Just beginning to explore using
solar..

**Comment by dalewegner** [comment]
Here is a real email to pass the code to. ***@***.***

I had read the solar discussion and it seemed like some challenges needed
to be solved.  I was not sure where to start with the sunrise condition and
powering up the board for a new day.  Had started more down the route of
putting the processor into a deep sleep rather than doing a cold start.

Thanks.
